module/VuFind/src/VuFind/View/Helper/Root/MakeLink.php

```diff
@@ -0,0 +1,94 @@
+<?php
+/**
+ * Make link view helper
+ *
+ * PHP version 7
+ *
+ * Copyright (C) Villanova University 2020.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 2,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+namespace VuFind\View\Helper\Root;
+
+use Laminas\View\Helper\AbstractHelper;
+
+/**
+ * Make link view helper
+ *
+ * @category VuFind
+ * @package  View_Helpers
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class MakeLink extends AbstractHelper
+{
+    /**
+     * Create an anchor tag (or just text if no href provided)
+     *
+     * @param string $text  Link contents
+     * @param string $href  Link destination
+     * @param array  $attrs Link attributes (associative array)
+     *
+     * @return string HTML for an anchor tag
+     */
+    public function __invoke($text, $href = null, $attrs = [])
+    {
+        if (!$href && !isset($attrs['href'])) {
+            $escHTML = $this->getView()->plugin('escapeHtml');
```

Comment:
Do we really want to escape HTML inside this helper, or should we rely on the incoming text being pre-escaped? I can imagine we might want to wrap a link around some other HTML, and that seems impossible with this setup. Alternatively, we could add a fourth "$escapeText" parameter that defaults to true, if we want to make escaping convenient and automatic while still offering a way around it for edge cases.

Reply:
I was wondering about the use case of wrapping HTML and I don't think this makeLink approach would work for that. It would have to be closer to the slots syntax:

```php
$this->wrapWithLink($check ? $url : false)->start();
  // All kinds of crazy record cover logic
$this->wrapWithLink()->end();
```

Reply:
I do want to avoid double escaping though...

Reply:
I agree that slots make more sense for large, complicated things... but I could also envision relatively lightweight use cases, like a sentence containing bold or italics. I think adding a parameter that allows prevention of double-escaping would be a pretty lightweight solution to cover that base.

Reply:
Also consider the fact that the text might be a translation string that allows html (e.g. add_to_favorites_html).

Reply:
Good point -- and in that situation, I would definitely prefer to simply translate the string outside of the view helper, and disable escaping (e.g. `$this->makeLink($this->translate(...), $link, [], false)`), rather than trying to add logic inside the helper to try to detect what to do; I think that would be unnecessarily complicated and error-prone.

Reply:
I added some tests to show how the escaping works with RecordLink. It is escaping but wouldn't be double escaping using record link.

Reply:
My question, though, is whether the existing RecordLink tests are not capturing cases where HTML escaping gets applied to the generated URL. What happens if you generate a link with RecordLink that has multiple GET parameters? How are the ampersands escaped? (Maybe we need to add another case to the existing RecordLinkTest to answer this question and document this behavior).

Reply:
See #2072 for further discussion of this topic; so far, I can't find a case that causes a problem here, but I want to be sure we do our due diligence in case there's a narrower edge case that we'll regret later.

Reply:
@crhallberg, it seems that the place we've landed on here is that the helper now ALWAYS escapes text inside the attributes on the `<a>` tag, and NEVER escapes the text within the `<a>` tag. I think this is reasonably flexible and appropriate, but wanted to be sure we were in agreement that it was the desired behavior. I've added some tests and comments in commit 
ce5c4d5 in an effort to clarify the situation.